compiler/rustc_mir/src/transform/large_enums.rs

```diff
@@ -0,0 +1,218 @@
+                        let mut data =
+                            vec![0; std::mem::size_of::<usize>() * num_variants as usize];
+                        data.copy_from_slice(unsafe { std::mem::transmute(&sizes[..]) });
+                        let alloc = interpret::Allocation::from_bytes(
+                            data,
+                            tcx.data_layout.ptr_sized_integer().align(&tcx.data_layout).abi,
+                        );
+                        let alloc = tcx.intern_const_alloc(alloc);
+                        let constant_vals = Constant {
+                            span,
+                            user_ty: None,
+                            literal: ConstantKind::Val(
+                                interpret::ConstValue::ByRef { alloc, offset: Size::ZERO },
+                                tmp_ty,
+                            ),
+                        };
+                        let rval = Rvalue::Use(Operand::Constant(box (constant_vals)));
+
+                        let const_assign = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (place, rval)),
+                        };
+
+                        // FIXME(jknodt) do I need to add a storage live here for this place?
+                        let discr_place = Place {
+                            local: patch.new_temp(tcx.types.usize, span),
+                            projection: List::empty(),
+                        };
+
+                        let store_discr = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (
+                                discr_place,
+                                Rvalue::Discriminant(*rhs),
+                            )),
+                        };
+
+                        // FIXME(jknodt) do I need to add a storage live here for this place?
+                        let size_place = Place {
+                            local: patch.new_temp(tcx.types.usize, span),
+                            projection: List::empty(),
+                        };
+
+                        let store_size = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (
+                                size_place,
+                                Rvalue::Use(Operand::Copy(Place {
+                                    local: discr_place.local,
+                                    projection: tcx
+                                        .intern_place_elems(&[PlaceElem::Index(size_place.local)]),
+                                })),
+                            )),
+                        };
+
+                        // FIXME(jknodt) do I need to add a storage live here for this place?
+                        let dst = Place {
+                            local: patch.new_temp(
+                                tcx.mk_ty(ty::RawPtr(ty::TypeAndMut {
```

Comment:
TyCtxt has convenience constructors for pointer types: `mk_imm_ptr` and `mk_mut_ptr`.